src/components/page-details.jsx

```diff
@@ -67,11 +67,17 @@ export default class PageDetails extends React.Component {
    * @param {Object} annotation
    */
   _annotateChange (fromVersion, toVersion, annotation) {
-    this.context.api.annotateChange(this.state.page.uuid, fromVersion, toVersion, annotation);
+    this.setState({ updating: true })
+    this.context.api.annotateChange(
```

Comment:
You will probably need to return this Promise, so that it is thenable in `_saveAnnotation ()` in `change-view.jsx`